### PR TITLE
Remove `qiskit.test`

### LINKED ISSUE #1244: qiskit.test is going to be removed in qiskit 1.0
Qiskit will remove `qiskit.test` in 1.0, which is planned to be released on Jan 31st, 2024 (see [issue](https://github.com/Qiskit/qiskit/issues/6862)). 

`qiskit-ibm-runtime` uses `qiskit.test` in various places, and they'd need to be updated before the release.

### PR DESCRIPTION
**Pull request by kt474**
### Details and comments
Fixes #1244

### CODE REVIEW

**Comment by kt474** [comment]
There is one instance of `qiskit.test` in one of the transpiler tests [here](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/test/unit/transpiler/passes/scheduling/control_flow_test_case.py#L18) where `canonicalize_control_flow` is imported, but it doesn't not need to be removed because it was kept in `qiskit/test`. You can see Jake's comment [here](https://github.com/Qiskit/qiskit/pull/11445#issue-2051101435)

File: qiskit_ibm_runtime/qiskit_runtime_service.py [hunk 1]
```diff
@@ -82,15 +82,22 @@ class QiskitRuntimeService(Provider):
...
 
         # Prepare inputs.
-        bell = ReferenceCircuits.bell()
         psi = RealAmplitudes(num_qubits=2, reps=2)
         H1 = SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)])
         theta = [0, 1, 1, 2, 3, 5]
 
+        # Bell Circuit
+        qr = QuantumRegister(2, name="qr")
+        cr = ClassicalRegister(2, name="qc")
```
Comment (jyu00) [inline qiskit_ibm_runtime/qiskit_runtime_service.py:91]: nit - the convention is `cr` for classical register

```suggestion
        cr = ClassicalRegister(2, name="cr")
```

File: qiskit_ibm_runtime/sampler.py [hunk 2]
```diff
@@ -46,16 +46,23 @@ class Sampler(BasePrimitive, BaseSampler):
...
-        from qiskit.test.reference_circuits import ReferenceCircuits
+        from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
         from qiskit_ibm_runtime import QiskitRuntimeService, Session, Sampler
 
         service = QiskitRuntimeService(channel="ibm_cloud")
-        bell = ReferenceCircuits.bell()
+
+        # Bell Circuit
+        qr = QuantumRegister(2, name="qr")
+        cr = ClassicalRegister(2, name="qc")
```
Comment (jyu00) [inline qiskit_ibm_runtime/sampler.py:56]:
```suggestion
        cr = ClassicalRegister(2, name="cr")
```

File: qiskit_ibm_runtime/session.py [hunk 3]
```diff
@@ -53,14 +53,22 @@ class Session:
 
     For example::
 
-        from qiskit.test.reference_circuits import ReferenceCircuits
+        from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
         from qiskit_ibm_runtime import Sampler, Session, Options
 
+        # Bell Circuit
+        qr = QuantumRegister(2, name="qr")
+        cr = ClassicalRegister(2, name="qc")
```
Comment (jyu00) [inline qiskit_ibm_runtime/session.py:61]:
```suggestion
        cr = ClassicalRegister(2, name="cr")
```

File: test/utils.py [hunk 4]
```diff
@@ -345,7 +344,19 @@ def bell_in_qobj(backend: IBMBackend, shots: int = 1024) -> QasmQobj:
...
+        transpile(bell(), backend=backend),
         backend=backend,
         shots=shots,
     )
+
+
+def bell():
+    """Return a Bell circuit."""
+    quantum_register = QuantumRegister(2, name="qr")
+    classical_register = ClassicalRegister(2, name="qc")
```
Comment (jyu00) [inline test/utils.py:356]:
```suggestion
    classical_register = ClassicalRegister(2, name="cr")
```

**Review by jyu00 — APPROVED** [review]

**Pull request merged**